### ISSUE TITLE
Crash when dragging an ammo box (AGM) 

### ISSUE THREAD

**Issue by AGroupOfBears** [usecforce/USEC_IssuesWiki#33]
When trying to drag some objects (ammo box) on the carrier (on the chronos ops server) causes a crash.

The client side RPT spams the following just before the crash.

```
18:23:02 Error: Weapon[rhs_m4_compm4]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
18:23:02 Error: Weapon[rhs_weap_m16a4_grip_acog_usmc]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
18:23:02 Error: Weapon[rhs_weap_m4a1_blockII_grip2_acog2_usmc]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
18:23:02 Error: Weapon[rhs_weap_mk18_grip2_eotech]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]

Exception code: C0000005 ACCESS_VIOLATION at 014AB25B
graphics:  D3D11
resolution:  1920x1080x32
Addons:
```

as well as a few other errors, the server RPT might yield better results.

This has been tested on the carrier as well as the mainland.

**Comment by jonpas** [comment]
Looks like an issue with ASDG Joint Rails, there is already a new version available. 

Can you go to editor and give it a test locally? Then disable ASDG JR and try it again.

**Comment by AGroupOfBears** [comment]
Locally I can drag a Virtual Arsenal just fine with join rails active.

the same result when join rails was not active.

I don't think Joint Rails is what's causing it, the problem is only present when using the AGM menu to drag an an ammo box. 

Dragging a tire/jerry can work fine though.

When done locally, the following code is spammed in the RPT but does not cause a crash

```
20:34:06 Error: Weapon[arifle_MXC_Holo_pointer_F]: max attachments slots count reached(maximum = 5)!  Attachment slot[asdg_OpticRail_MX] for weapon's proxy[\A3\data_f\proxies\weapon_slots\TOP]
20:34:06 Error: Weapon[arifle_MXC_Holo_F]: max attachments slots count reached(maximum = 5)!  Attachment slot[asdg_OpticRail_MX] for weapon's proxy[\A3\data_f\proxies\weapon_slots\TOP]
20:34:06 Error: Weapon[arifle_MXC_ACO_F]: max attachments slots count reached(maximum = 5)!  Attachment slot[asdg_OpticRail_MX] for weapon's proxy[\A3\data_f\proxies\weapon_slots\TOP]
20:34:06 Error: Weapon[arifle_MXC_ACO_pointer_snds_F]: max attachments slots count reached(maximum = 5)!  Attachment slot[asdg_OpticRail_MX] for weapon's proxy[\A3\data_f\proxies\weapon_slots\TOP]
20:34:06 Error: Weapon[rhs_weap_m4_carryhandle]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
20:34:06 Error: Weapon[rhs_m4_compm4]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
20:34:06 Error: Weapon[rhs_weap_m4_carryhandle_pmag]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
20:34:06 Error: Weapon[rhs_weap_m16a4_grip_acog_usmc]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
20:34:06 Error: Weapon[rhs_m4_acog_usmc]: max attachments slots count reached(maximum = 5)!  Attachment slot[UnderBarrelSlot] for weapon's proxy[\A3\Data_F_Mark\Proxies\Weapon_Slots\UNDERBARREL]
```

**Comment by jonpas** [comment]
Then I don't really know if we can reliably fix it. I won't have time to work on anything until tomorrow afternoon anyways, go ahead and try to find the cause if you want to.

EDIT: Checked dedicated server RPT and there is nothing there, must be something BIS changed in 1.42 update that causes a crash, and it happens when pressing the Drag button.

**Comment by jonpas** [comment]
Possibly fixed for 0.9.1.

**Comment by jonpas** [comment]
Confirmed fixed in jonpas/AGM@b61c1eabc2c94bc7bcd9a4f3b511ac48afefebdc
